hummingbot/client/config/config_helpers.py

```diff
@@ -191,21 +189,36 @@ def get_strategy_starter_file(strategy: str) -> Callable:
         logging.getLogger().error(e, exc_info=True)
 
 
-def load_required_configs(*args) -> OrderedDict:
-    """
-    Go through `in_memory_config_map`, `strategy_config_map`, and `global_config_map` in order to list all of the
-    config settings required by the bot.
-    """
-    from hummingbot.client.config.in_memory_config_map import in_memory_config_map
-    current_strategy = global_config_map.get("strategy").value
-    current_strategy_file_path = in_memory_config_map.get("strategy_file_path").value
-    if current_strategy is None or current_strategy_file_path is None:
-        return _merge_dicts(in_memory_config_map)
-    else:
-        strategy_config_map = get_strategy_config_map(current_strategy)
-        # create an ordered dict where `strategy` is inserted first
-        # so that strategy-specific configs are prompted first and populate required_exchanges
-        return _merge_dicts(in_memory_config_map, strategy_config_map, global_config_map)
+def load_required_configs(strategy_name) -> OrderedDict:
+    strategy_config_map = get_strategy_config_map(strategy_name)
+    # create an ordered dict where `strategy` is inserted first
+    # so that strategy-specific configs are prompted first and populate required_exchanges
+    return _merge_dicts(strategy_config_map, global_config_map)
+
+
+def strategy_type_from_file(file_path: str) -> str:
+    strategies = get_strategy_list()
+    for strategy in strategies:
+        if strategy_matches_file(file_path, strategy):
+            return strategy
+
+
+def strategy_matches_file(file_path, strategy) -> bool:
```

Comment:
This method currently works, but in the future we plan to support users creating their own strategies or loading other custom strategies. Would defining a `strategy` param be better?

Reply:
We simply need a kinda hard coded value in the file, it doesn't have to a config variable, just another entry in the template file will do. Though this will not be backward compatible as older version config files will not have this field. 
I'm fine with either approaches.

Reply:
Let's require a `strategy` param in each config file that specifies the strategy name.

Since we're making so many changes to the client, the old config files won't be backward compatible with this new release anyway.

Reply:
Yep, agree, probably will easier for other developers to understand the code base this way too.
I have made the changes to all strategy config files and their templates. Also fixed the problem with creating Arbitrage and XEMM strategy configs.